packages/protobuf/src/from-json.ts

```diff
@@ -477,17 +445,8 @@ function scalarFromJson(
 function scalarFromJson(
   field: DescField & { scalar: ScalarType },
   json: JsonValue,
-  nullAsZeroValue: false,
-): ScalarValue | typeof tokenNull | JsonValue;
-function scalarFromJson(
-  field: DescField & { scalar: ScalarType },
-  json: JsonValue,
-  nullAsZeroValue: boolean,
 ): ScalarValue | typeof tokenNull | JsonValue {
   if (json === null) {
-    if (nullAsZeroValue) {
-      return scalarZeroValue(field.scalar, false);
-    }
```

Comment:
All calls that passed `nullAsZeroValue` as `true` also guaranteed that `json` was not `null`.